### PR TITLE
Add secret map to votemap

### PR DESCRIPTION
**Pull request by VigersRay**
## About the PR

Are you tired of always choosing one map in voting? The solution is found, the secret map is in the vote. Checked on one of the servers, with a high probability everyone will choose the secret.

**Media**

Screenshot_76
Screenshot_77

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl: VigersRay
- add: Added secret map to votemap

### CODE REVIEW

**Comment by ficcialfaint** [comment]
I got five cards (it's called maps not cards 🃏) image

File: Content.Server/Voting/Managers/VoteManager.DefaultVotes.cs [hunk 1]
```diff
@@ -158,7 +158,13 @@ private void CreatePresetVote(IPlayerSession? initiator)
 
         private void CreateMapVote(IPlayerSession? initiator)
         {
-            var maps = _gameMapManager.CurrentlyEligibleMaps().ToDictionary(map => map, map => map.MapName);
+            var maps = new Dictionary<string, GameMapPrototype>();
+            var eligibleMaps = _gameMapManager.CurrentlyEligibleMaps().ToList();
+            maps.Add("Secret", _random.Pick(eligibleMaps));
```
Comment (ficcialfaint) [inline Content.Server/Voting/Managers/VoteManager.DefaultVotes.cs:163]: This should be fluent I think

Comment (VigersRay) [reply Content.Server/Voting/Managers/VoteManager.DefaultVotes.cs:163]: I don’t think there is a check that the secret won and the check goes just along the line. So it's impossible to translate

Comment (ficcialfaint) [reply Content.Server/Voting/Managers/VoteManager.DefaultVotes.cs:163]: huh?

Comment (VigersRay) [reply Content.Server/Voting/Managers/VoteManager.DefaultVotes.cs:163]: The vote always shows the code name of the cards and it is not translated, why then translate the secret version? In any case, I have no idea how to solve this problem.

Comment (ficcialfaint) [reply Content.Server/Voting/Managers/VoteManager.DefaultVotes.cs:163]: it's fucking maps not cards smh.
You can do 
```cs
if (title == Loc.GetString("secret-map")) {
```

Comment (ficcialfaint) [reply Content.Server/Voting/Managers/VoteManager.DefaultVotes.cs:163]: The fact that your if statement is basically compares names is a much huger crutch. Ion see anything bad with comparing fluent strings. You didn't explain why you think so

Comment (VigersRay) [reply Content.Server/Voting/Managers/VoteManager.DefaultVotes.cs:163]: > The fact that your if statement is basically compares names is a much huger crutch. Ion see anything bad with comparing fluent strings. You didn't explain why you think so
In any case, I would like to see an option suggestion without the if (title = "Secret") check

Comment (VigersRay) [reply Content.Server/Voting/Managers/VoteManager.DefaultVotes.cs:163]: > The fact that your if statement is basically compares names is a much huger crutch. Ion see anything bad with comparing fluent strings. You didn't explain why you think so

In any case, I would like to see an option suggestion without the if (title = "Secret") check

**Comment by lzk228** [comment]
when

File: Content.Server/Voting/Managers/VoteManager.DefaultVotes.cs [hunk 2]
```diff
@@ -184,17 +190,24 @@ private void CreateMapVote(IPlayerSession? initiator)
...
                     picked = (GameMapPrototype) args.Winner;
-                    _chatManager.DispatchServerAnnouncement(
-                        Loc.GetString("ui-vote-map-win", ("winner", maps[picked])));
+                    title = maps.FirstOrDefault(x => x.Value == picked).Key;
+                    _chatManager.DispatchServerAnnouncement(Loc.GetString("ui-vote-map-win",
+                        ("winner", title)));
+                }
+                if (title == Loc.GetString("ui-vote-secret-map"))
+                {
+                    _chatManager.DispatchServerAnnouncement(Loc.GetString("ui-vote-secret-win"));
```
Comment (AJCM-git) [inline Content.Server/Voting/Managers/VoteManager.DefaultVotes.cs:210]: This seems a lil hacky, I would just leave it as it was, having it generate "Secret won the map vote!" is clear enough and better than using ftl strings as conditions

File: Content.Server/Voting/Managers/VoteManager.DefaultVotes.cs [hunk 3]
```diff
@@ -158,7 +158,13 @@ private void CreatePresetVote(IPlayerSession? initiator)
...
         private void CreateMapVote(IPlayerSession? initiator)
         {
-            var maps = _gameMapManager.CurrentlyEligibleMaps().ToDictionary(map => map, map => map.MapName);
+            var maps = new Dictionary<string, GameMapPrototype>();
+            var eligibleMaps = _gameMapManager.CurrentlyEligibleMaps().ToList();
+            maps.Add(Loc.GetString("ui-vote-secret-map"), _random.Pick(eligibleMaps));
+            foreach (var map in eligibleMaps)
+            {
+                maps.Add(map.MapName, map);
+            }
```
Comment (AJCM-git) [inline Content.Server/Voting/Managers/VoteManager.DefaultVotes.cs:167]: Is there something wrong with
```suggestion
            var maps = _gameMapManager.CurrentlyEligibleMaps().ToDictionary(map => map, map => map.MapName);
            maps.Add(Loc.GetString("ui-vote-secret-map"), _random.Pick(eligibleMaps));
```

**Review by AJCM-git — CHANGES_REQUESTED** [review]

**Comment by AJCM-git** [comment]
@VigersRay You coming back to this?

File: Content.Server/Voting/Managers/VoteManager.DefaultVotes.cs (hunk 3 above)
Comment (Partmedia) [reply Content.Server/Voting/Managers/VoteManager.DefaultVotes.cs:167]: The submitter was probably not aware of ToDictionary(). Committing suggestion.

File: Content.Server/Voting/Managers/VoteManager.DefaultVotes.cs [hunk 4]
```diff
@@ -158,13 +158,8 @@ private void CreatePresetVote(IPlayerSession? initiator)
 
         private void CreateMapVote(IPlayerSession? initiator)
         {
-            var maps = new Dictionary<string, GameMapPrototype>();
-            var eligibleMaps = _gameMapManager.CurrentlyEligibleMaps().ToList();
+            var maps = _gameMapManager.CurrentlyEligibleMaps().ToDictionary(map => map, map => map.MapName);
```
Comment (Partmedia) [inline Content.Server/Voting/Managers/VoteManager.DefaultVotes.cs:161]: Ooops the key and values need to be switched.

**Comment by AJCM-git** [comment]
re-open once you come back to this

**Pull request closed without merging**